### PR TITLE
clarifications on FI_ORDER_NONE

### PR DESCRIPTION
**Pull request by sayantansur**
fixes issue #2198.

Signed-off-by: Sayantan Sur <sayantan.sur@intel.com>

### CODE REVIEW

**Comment by lanl-ompi** [comment]
bot:retest

**Comment by lanl-ompi** [comment]
Test FAILed.

File: man/fi_endpoint.3.md [hunk 1]
```diff
@@ -1207,6 +1208,16 @@ Operations that complete with an error increment the error counter
...
 
+As mentioned in fi_getinfo(3), the ep_attr structure can be used to
+query providers that support various endpoint attributes. fi_getinfo
+can return provider info structures that can support the minimal set
+of requirements (such that the application maintains correctness).
+However, it can also return provider info structures that exceed
+application requirements. As an example, consider an application
+requesting msg_order as FI_ORDER_NONE. The resulting output from
+getinfo may have all the ordering bits set. The application can reset
+the ordering bits it does not require before creating the endpoint.
```
Comment (j-xiong) [inline man/fi_endpoint.3.md]: How about adding something like this to the end: "but that's optional and a provider is not required to adapt to the relaxed requirement."

Comment (sayantansur) [inline man/fi_endpoint.3.md]: how about the following addition?

Comment (j-xiong) [inline man/fi_endpoint.3.md]: Sounds good.

**Comment by hppritcha** [comment]
jenkins/nersc fixups.
bot:retest

**Pull request merged**